### DISCUSSION TITLE
How can I execute a function after this.selection.startSelecting(evt); is completed in order to remove selected elements based on attributes?

### DISCUSSION THREAD

**Discussion by goyaljai1** [clientIO/joint#2785] [Q&A] [answered]
### Introduction

I’m using a selection tool in my application that initiates with this.selection.startSelecting(evt); when a user begins the selection process. I want to execute a cleanup function once the selection is complete to remove certain elements based on their attribute properties.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
You can listen to the next collection reset event:
```ts
paper.on('blank:pointerdown', (evt) => {
    selection.startSelecting(evt);
    selection.collection.once('reset', () => graph.removeCells(selection.collection.toArray());
});
```
